frontend/src/Components/UserHomePage/UserHomePage.jsx

```diff
@@ -13,6 +13,10 @@ const HomePage = () => {
     localStorage.removeItem("user");
```

Comment:
It would be cleaner to have all 'cleanup' related functions in a separate function. e.g. cleanUpUser containing the state update + local storage update. Not blocking -- let's clean these up in a separate PR